src/inventorymanager.cpp

```diff
@@ -533,20 +533,42 @@ void IMoveAction::clientApply(InventoryManager *mgr, IGameDef *gamedef)
 	// Optional InventoryAction operation that is run on the client
 	// to make lag less apparent.
 
-	Inventory *inv_from = mgr->getInventory(from_inv);
-	Inventory *inv_to = mgr->getInventory(to_inv);
-	if(!inv_from || !inv_to)
+	/*
+		Disable moving items out of craftpreview
+	*/
+	if (from_list == "craftpreview")
```

Comment:
without it, if you place items while picking them up from the craft preview, you end up getting 2 stacks in your hands(!), very confusing.

I don't understand why moving out of the craftpreview is blocked, though.